src/main/java/com/nutomic/syncthingandroid/preferences/WifiSsidPreference.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    public WifiSsidPreference(Context context) {
+        super(context);
+    }
+
+    /**
+     * Show the dialog if WiFi is available and configured networks can be loaded.
+     * Otherwise will display a Toast requesting to turn on WiFi.
+     *
+     * On opening of the dialog, will also remove any SSIDs from the set that have been removed
+     * by the user in the WiFi manager. This change will be persisted only if the user changes
+     * any other setting
+     */
+    @Override
+    protected void showDialog(Bundle state) {
+        WifiConfiguration[] networks = loadConfiguredNetworksSorted();
+        if (networks != null) {
+            Set<String> selected = getSharedPreferences().getStringSet(getKey(), new HashSet<String>());
+            // from JavaDoc: Note that you must not modify the set instance returned by this call.
+            // therefore required to make a defensive copy of the elements
+            selected = new HashSet<>(selected);
+            CharSequence[] all = extractSsid(networks, false);
+            filterRemovedNetworks(selected, all);
+            setEntries(extractSsid(networks, true)); // display without surrounding quotes
+            setEntryValues(all); // the value of the entry is the SSID "as is"
+            setValues(selected); // the currently selected values (without meanwhile deleted networks)
+            super.showDialog(state);
+        } else {
+            Toast.makeText(getContext(), R.string.sync_only_wifi_ssids_wifi_turn_on_wifi, Toast.LENGTH_LONG).show();
+        }
+    }
+
+    /**
+     * Removes any network that is no longer saved on the device. Otherwise it will never be
+     * removed from the allowed set by MultiSelectListPreference.
+     */
+    private void filterRemovedNetworks(Set<String> selected, CharSequence[] all) {
+        HashSet<CharSequence> availableNetworks = new HashSet<>(Arrays.asList(all));
+        selected.retainAll(availableNetworks);
+    }
+
+    /**
+     * Converts WiFi configuration to it's string representation, using the SSID.
+     *
+     * It can also remove surrounding quotes which indicate that the SSID is an UTF-8
+     * string and not a Hex-String, if the strings are intended to be displayed to the
+     * user, who will not expect the quotes.
+     *
+     * @param configs the objects to convert
+     * @param stripQuotes if to remove surrounding quotes
+     * @return the formatted SSID of the wifi configurations
+     */
+    private CharSequence[] extractSsid(WifiConfiguration[] configs, boolean stripQuotes) {
+        CharSequence[] result = new CharSequence[configs.length];
+        for (int i = 0; i < configs.length; i++) {
+            // WiFi SSIDs can either be UTF-8 (encapsulated in '"') or hex-strings
+            if (stripQuotes)
+                result[i] = configs[i].SSID.replaceFirst("^\"", "").replaceFirst("\"$", "");
+            else
+                result[i] = configs[i].SSID;
+        } // for
```

Comment:
It's so natural to my team's style, that it took me (more than) a few seconds to realize what you were referring to :)